lib/AST/Stmt.cpp

```diff
@@ -321,9 +326,18 @@ SourceLoc StmtConditionElement::getStartLoc() const {
   case StmtConditionElement::CK_Availability:
     return getAvailability()->getStartLoc();
   case StmtConditionElement::CK_PatternBinding:
+    SourceLoc Start;
     if (IntroducerLoc.isValid())
-      return IntroducerLoc;
-    return getPattern()->getStartLoc();
+      Start = IntroducerLoc;
+    else
+      Start = getPattern()->getStartLoc();
+    
+    SourceLoc End = getInitializer()->getEndLoc();
```

Comment:
Then it would recheck the kind, we could just make a static function for this case.